### PR TITLE
fix: strongly type get_configure_view

### LINKED ISSUE #50: error with get_configure_view
With sentry  24.8.0

```
Traceback (most recent call last):
  File "/.venv/bin/sentry", line 4, in <module>
    raise SystemExit(main())
                     ^^^^^^
  File "/usr/src/sentry/src/sentry/runner/main.py", line 149, in main
    func(**kwargs)
  File "/.venv/lib/python3.11/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.11/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.11/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.11/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.11/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.11/site-packages/click/decorators.py", line 33, in new_func
    return f(get_current_context(), *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/sentry/src/sentry/runner/main.py", line 98, in inner
    ctx.forward(django)
  File "/.venv/lib/python3.11/site-packages/click/core.py", line 804, in forward
    return __self.invoke(__cmd, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.11/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.11/site-packages/click/decorators.py", line 33, in new_func
    return f(get_current_context(), *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/sentry/src/sentry/runner/decorators.py", line 34, in inner
    configure()
  File "/usr/src/sentry/src/sentry/runner/__init__.py", line 33, in configure
    _configure(ctx, py, yaml, skip_service_validation)
  File "/usr/src/sentry/src/sentry/runner/settings.py", line 121, in configure
    initialize_app(
  File "/usr/src/sentry/src/sentry/runner/initializer.py", line 366, in initialize_app
    django.setup()
  File "/.venv/lib/python3.11/site-packages/django/__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/.venv/lib/python3.11/site-packages/django/apps/registry.py", line 124, in populate
    app_config.ready()
  File "/.venv/lib/python3.11/site-packages/oidc/apps.py", line 10, in ready
    from .provider import OIDCProvider
  File "/.venv/lib/python3.11/site-packages/oidc/provider.py", line 17, in <module>
    from .views import FetchUser, OIDCConfigureView
  File "/.venv/lib/python3.11/site-packages/oidc/views.py", line 3, in <module>
    from sentry.auth.view import AuthView, ConfigureView
ImportError: cannot import name 'ConfigureView' from 'sentry.auth.view' (/usr/src/sentry/src/sentry/auth/view.py)

```

### PR DESCRIPTION
**Pull request by reneluria**
cf. https://github.com/getsentry/sentry/pull/75395

to be compatible with 24.8.0

fixes #50

### CODE REVIEW

**Comment by max-wittig** [comment]
Looks good to me. I will try it out soon. We have not yet upgraded to 24.8.0, but will do so soon!

**Comment by dlouzan** [comment]
@reneluria Thanks for the contribution! I'm currently checking this on our stage system, we also need it for sentry 24.8.0 👍

File: oidc/provider.py [hunk 1]
```diff
@@ -37,7 +46,7 @@ def get_authorize_params(self, state, redirect_uri):
 
 
 class OIDCProvider(OAuth2Provider):
-    name = ISSUER
+    name = ISSUER if ISSUER else "oidc"
```
Comment (dlouzan) [inline oidc/provider.py:49]: @reneluria Could you elaborate on this change? We provide a way to customize the name via the `OIDC_ISSUER` config, as can be seen in the readme and constants.py.

/cc @max-wittig

File: oidc/views.py [hunk 2]
```diff
@@ -1,7 +1,15 @@
...
 
-from sentry.auth.view import AuthView, ConfigureView
+from django.http import HttpRequest
+from rest_framework.response import Response
+
+from sentry.auth.services.auth.model import RpcAuthProvider
+from sentry.auth.view import AuthView
 from sentry.utils import json
+from sentry.organizations.services.organization.model import RpcOrganization
+from sentry.plugins.base.response import DeferredResponse
```
Comment (dlouzan) [inline oidc/views.py:12]: I kwow we should have an auto-formatter here, could you re-order to keep the abc? 😄

File: oidc/views.py [hunk 3]
```diff
@@ -52,17 +60,18 @@ def dispatch(self, request, helper):
...
+    request: HttpRequest, organization: RpcOrganization, auth_provider: RpcAuthProvider
+) -> DeferredResponse:
+    config = auth_provider.config
+    if config.get("domain"):
+        domains: list[str] | None
+        domains = [config["domain"]]
+    else:
+        domains = config.get("domains")
+    return DeferredResponse(
+        "oidc/configure.html", {"provider_name": ISSUER or "", "domains": domains or []}
```
Comment (dlouzan) [inline oidc/views.py:73]:
```suggestion
        "oidc/configure.html",
        {"provider_name": ISSUER or "", "domains": domains or []},
```

Just for consistency with the previous code.

**Review by dlouzan — CHANGES_REQUESTED** [review]
@reneluria Thanks, just did an initial pass, please take a look.

**Comment by dlouzan** [comment]
@reneluria We will also need a fix of the test setup to be able to run the test suite. Let me know if you're willing to take care of the required work, or we should take over.

File: oidc/provider.py (hunk 1 above)
Comment (dlouzan) [reply oidc/provider.py:49]: @reneluria I just had a chat with @max-wittig and we also tested the current behaviour without setting the `OIDC_ISSUER`, everything works. Please remove this guard and use the `OIDC_ISSUER` configuration with an appropriate value if you want to use `"oidc"` as your name. Thanks!

File: oidc/views.py (hunk 2 above)
Comment (max-wittig) [reply oidc/views.py:12]: Let's fix that with an autoformatter later.

File: oidc/provider.py (hunk 1 above)
Comment (reneluria) [reply oidc/provider.py:49]: Ok, let's do this

File: oidc/views.py (hunk 3 above)
Comment (reneluria) [reply oidc/views.py:73]: ok

**Comment by dlouzan** [comment]
@reneluria Thanks for addressing the open threads, we'll be merging this one now as our internal tests worked. There will be a new release once we fix the test setup, which we are doing in #51.

**Review by dlouzan — APPROVED** [review]

**Pull request merged**